Sources/SwiftCompilerPluginMessageHandling/PluginMacroExpansionContext.swift

```diff
@@ -170,19 +170,35 @@ class SourceManager {
     from startKind: PositionInSyntaxNode = .afterLeadingTrivia,
     to endKind: PositionInSyntaxNode = .beforeTrailingTrivia
   ) -> SourceRange? {
-    guard let base = self.knownSourceSyntax[node.root.id] else {
+    if let base = self.knownSourceSyntax[node.root.id] {
+      return range(node.position(at: startKind)..<node.position(at: endKind), in: base)
+    } else {
       return nil
     }
```

Comment:
I feel like I’m missing something. Can’t this call

```suggestion
    return range(node.position(at: startKind)..<node.position(at: endKind), in: node)
```

And then we wouldn’t need the version that takes a `KnownSourceSyntax` at all.